snaps_openstack/provision/openstack/plugin/kolla_impl/kolla_utils.py

```diff
@@ -449,6 +418,40 @@ def __create_global(config, git_branch, pull_from_hub):
                 filedata = filedata.replace('kolla_external_vip_interface: '+'"'+external_interface+'"',
                                             'kolla_external_vip_interface: "dpdk_bridge"')
 
+    hosts = config.get(consts.OPENSTACK).get(consts.HOSTS)
+    gateway = ""
+    netmask = ""
+    
+    for j in range(len(hosts)):
+        interfaces = hosts[j].get(consts.HOST).get(consts.INTERFACES)
+        node_type = hosts[j].get(consts.HOST).get(consts.NODE_TYPE)
+        if 'controller' in node_type or len(hosts) == 1:
+            for i in range(len(interfaces)):
```

Comment:
Why not use instead:
for name in interfaces:

You can then remove line 430 too